diff/diff.go

```diff
@@ -0,0 +1,276 @@
+package diff
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/fatih/color"
+)
+
+// Opt is an option type that can be passed to New.
+type Opt func(Differ) Differ
+
+// WithFormat returns an Opt that wraps up differences
+// using a format string.  The format should contain
+// one '%s' to add the difference string in.
```

Comment:
Eh, I thought about it.  We'd have to change `diff.New` to return an error, and that's kinda weird because it's just a diffing tool.  If the format is wrong, `fmt.Sprintf` does a pretty good job of making an ugly-but-still-informative output that warns users.

Reply:
SGTM